src/meson.build

```diff
@@ -0,0 +1,74 @@
+dunst_depends = [
+	cairo,
+	glib,
+	gio,
+	gdk_pixbuf,
+	pangocairo,
+	systemd,
+	libnotify,
+	realtime,
+	math,
+]
+
+dunst_src_files = files(
+	'dbus.c', 
+	'draw.c', 
+	'dunst.c', 
+	'icon-lookup.c', 
+	'icon.c', 
+	'ini.c', 
+	'input.c', 
+	'log.c', 
+	'markup.c', 
+	'menu.c', 
+	'notification.c', 
+	'option_parser.c', 
+	'output.c', 
+	'queues.c', 
+	'rules.c', 
+	'settings.c', 
+	'utils.c', 
+)
+
+if x11_support
+	add_global_arguments('-DENABLE_X11', language: 'c')
```

Comment:
Why add_global here? Previously you used add_project -- which makes more sense anyway, in most cases.

Reply:
I would also suggest moving this to the top-level file anyway... you already have generic option checking code, why do this inside a subdir during target creation?